lib/ABI/Definition.cpp

```diff
@@ -318,12 +326,12 @@ naturalAlignment(const abi::Definition &ABI,
     revng_abort();
   }
 
-  VH.setAlignment(&Type, Alignment);
+  VH.setAlignment(&Type, Result);
```

Comment:
I am having second thoughts about this.

If I remember correctly, the whole reason alignment stuff got shoe-horned in here was because this whole thing in in a desperate need of a rework and piling stuff in would only make it happen faster.

As for your immediate concern, I don't see much downside against just doing:
```diff
private:
  struct AlignmentInfo;
-  std::map<const model::Type *, AlignmentInfo> AlignmentCache;
+  std::map<std::pair<const model::Type *, model::ABI::Values>, AlignmentInfo> AlignmentCache;

public:
-  void setAlignment(const model::Type *T, AlignmentInfo NewValue);
+  void setAlignment(abi::Definition const &ABI, const model::Type *T, AlignmentInfo NewValue);
-  std::optional<AlignmentInfo> alignment(const model::Type *T);
+  std::optional<AlignmentInfo> alignment(abi::Definition const &ABI, const model::Type *T);
```

And leaving it to future us to deal with the problem properly. At the very least it beats duplicating this already awful data-structure into two.

Reply:
1. Is there someone who cares about alignment out of `revngABI`?
   If not, there's no reason it should be in `revngModel`.
2. Correct me if I'm wrong, but the users of alignment-related methods of `VerifyHelper`, only use those methods of `VerifyHelper`. It's not like they care about other features of `VerifyHelper`.
   If this is the case, I don't see why we can't entirely decouple the thing, pass around a `AlignmentCache` object (instead of passing around `VerifyHelper`) and be done with it.

It sounds like a really simple change that decouples `revngModel` from `revngABI`.

Reply:
I'm not arguing that we don't want to decouple those - because we do. What I'm afraid of is that if we do so now, when `VerifyHelper` gets reworked, its abi-related equivalent will stay as it is now.

Reply:
Done, see the new `ABI: decouple alignment from model::VerifyHelper` commit and the updated `ABI: improve unnatural alignment support`.